### PR TITLE
Implement a new options page for Code style features.

### PR DESCRIPTION
**Pull request by balajikris**
Part of #9155

**_Summary of changes:_**

1. Implements a new options page for code styles. 
2. Brings the new use var code style into this page.
3. Brings existing code styles into this page.

**_Before:_**

image

**_With use var:_**
image

**_Latest (this PR):_**<sup>1</sup>

image

<sup>1</sup> Notification is empty/not shown for codestyles that don't yet implement notification preferences.

### CODE REVIEW

**Comment by balajikris** [comment]
For review: @CyrusNajmabadi  @dpoeschl @DustinCampbell  and @dotnet/roslyn-ide 

Admittedly, this is cutting very close to the last date for checking into preview before it is ask mode. Given this, my current strategy is to immediately address any merge-blocking or concerning feedback and continuing to address the rest of the feedback in ask mode or thereafter. 

**_some items that are not yet implemented but will be added in future iterations_:**

1. better keyboard support
2. high contrast support
3. screen reader friendliness
4. high DPI et al.

**Comment by balajikris** [comment]
retest vsi please

**Comment by balajikris** [comment]
@dotnet-bot retest prtest/win/vsi/p3 please
// Previous failure: http://dotnet-ci.cloudapp.net/job/dotnet_roslyn_prtest_win_vsi_p3/353/
// Retest reason: jenkins machine went offline during build.

**Comment by CyrusNajmabadi** [comment]
LGTB

File: src/VisualStudio/Core/Impl/Options/GridOptionPreviewControl.xaml [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+            AutoGenerateColumns="False"
+            CanUserReorderColumns="False"
+            CanUserSortColumns="False"
+            CanUserAddRows="False"
+            CanUserDeleteRows="False"
+            CanUserResizeRows="False"
+            CanUserResizeColumns="False"
+            IsReadOnly="True"
+            BorderThickness="1"
+            BorderBrush="Gray"
```
Comment (Pilchie) [inline src/VisualStudio/Core/Impl/Options/GridOptionPreviewControl.xaml]: Is this right for High Contrast, etc?  Is there a VS Brush we should be using?

**Comment by Pilchie** [comment]
:+1:

In the screenshot, there is no preview - is that because no option is selected?  Should we have some placeholder text in that case?

**Comment by balajikris** [comment]
responding to @Pilchie 's comments
> Is this right for High Contrast, etc?
totally did not do anything for high contrast. I'll address it in smaller changes, suitable for ask mode :wink: 

> In the screenshot, there is no preview - is that because no option is selected? Should we have some placeholder text in that case?
yes, you're right about the reason why there is no preview. In this case, we normally show preview for the first item, but I don't have that code working yet, i instead have a todo. We can also do some custom text there. Will address in followup.

**Comment by balajikris** [comment]
Rebased on top of @brettfo 's changes and resolved merge conflicts, also brought his new qualify this. codestyle choices to the new UI.

**Pull request merged**